citro3d/examples/triangle.rs

```diff
@@ -86,22 +91,29 @@ fn main() {
             break;
         }
 
-        instance.render_frame_with(|instance| {
-            let clear_color: u32 = 0x7F_7F_7F_FF;
-            render_target.clear(ClearFlags::ALL, clear_color, 0);
+        let mut render_to = |target: &mut Target| {
+            instance.render_frame_with(|instance| {
+                instance
+                    .select_render_target(target)
+                    .expect("failed to set render target");
 
-            instance
-                .select_render_target(&render_target)
-                .expect("failed to set render target");
+                let clear_color: u32 = 0x7F_7F_7F_FF;
+                target.clear(ClearFlags::ALL, clear_color, 0);
+                scene_render(uloc_projection.into(), &projection);
+            });
+        };
 
-            scene_render(uloc_projection.into(), &projection);
-        });
+        render_to(&mut top_target);
+        render_to(&mut bottom_target);
     }
 
-    scene_exit(vbo_data);
+    // explicit drop to ensure the vbo_data lives long enough for the render code
+    // to reference it. This hopefully won't be necessary anymore if we can use
+    // lifetimes to manage the buffers instead of passing raw ptrs
+    drop(vbo_data);
```

Comment:
Note that this isn't needed. Rust won't change the drop order based on lifetimes (i.e. NLL doesn't affect drop order or timing)

Reply:
Huh, TIL. I tried looking around for an explicit reference on this, but mostly just found the [drop scopes reference](https://doc.rust-lang.org/reference/destructors.html#drop-scopes) which doesn't seem to mention anything about NLL specifically (I guess because it's irrelevant).

Anyway, I still want to do a proper safe wrapper as you alluded to in your other comment, so maybe I'll do that in another PR after this.